### ISSUE TITLE
Support for RAID

### ISSUE THREAD

**Issue by barryspearce** [Chrysostomus/aif-dev#204]
Servers are frequently configured using bootable UEFI + RAID + LVM.
This was possible under Calamares. Since the demise of that, and the total lack of RAID support in the GUI installers for Manjaro Manjaro Architect should support RAID creation.

At the current time disk partitioning and RAID set creation effectively must be done before starting Manjaro Architect. 
RAID devices are not shown under the LVM options.

As the software is not RAID aware it shows LVM devices (/dev/mapper) multiple times - one for each disk!

RAID devices should be shown for mounting file systems as well (so you can mount a RAID /boot). 

Grub installation (or other bootloader) should also take account of multiple boot partitions when using mirroring.
EFI must also be configured for multiple entries in bios when using mirroring.

**Comment by Chrysostomus** [comment]
Yes, this should be implemented, along with the btrfs style raid. I'm thinking of raid submenu in the preparation menu, which would branch out to lvm, mdadm and btrfs sections. It is one of the features that should be included in 1.0 release. 

Trouble is, I currently lack the hardware and experience to implement this feature. I have never used raid of any kind. So it might take a good while unless we find someone else to do it.

**Comment by barryspearce** [comment]
All my systems here (including desktops!) are UEFI + RAID + LVM systems.
Let me look at the source and see if its something I can help with...

**Comment by Chrysostomus** [comment]
It is kinda messy at the moment, two big changes without code cleaning yet. Until-disk.sh might be a good place to start.

**Comment by barryspearce** [comment]
OK so its all in shell script. 
RAID isn't a big deal hardware wise - just two disks with partitions of the same size is all you need to get started with mirroring (which is probably the most common config on smaller systems).

**Comment by gour** [comment]
Any progress on this? It's one of the things which still keeps me on Debian where it's easy to do LVM+RAID1 install...

**Comment by Chrysostomus** [comment]
https://gitlab.manjaro.org/applications/manjaro-architect/issues/204

**Comment by gour** [comment]
@Chrysostomus Thank a lot!
